src/alert-manager/src/cert-expiration-checker/send_alert.py

```diff
@@ -0,0 +1,60 @@
+from datetime import timezone, datetime, timedelta
+import logging
+import os
+import requests
+
+ALERT_PREFIX = "/alert-manager/api/v1/alerts"
+alertResidualDays = int(os.environ.get('ALERT_RESIDUAL_DAYS'))
+
+def enable_request_debug_log(func):
+    def wrapper(*args, **kwargs):
+        requests_log = logging.getLogger("urllib3")
+        level = requests_log.level
+        requests_log.setLevel(logging.DEBUG)
+        requests_log.propagate = True
+
+        try:
+            return func(*args, **kwargs)
+        finally:
+            requests_log.setLevel(level)
+            requests_log.propagate = False
+
+    return wrapper
+
+@enable_request_debug_log
+def send_alert(pai_url: str, certExpirationInfo: str):
+    trigger_time = str(datetime.now(timezone.utc).date())
+    post_url = pai_url.rstrip("/") + ALERT_PREFIX
+    alerts = []
+    alert = {
+        "labels": {
+            "alertname": "k8s cert expiration",
+            "severity": "warn",
+            "trigger_time": trigger_time,
+            "annotations": certExpirationInfo,
+        },
+        "generatorURL": "alert/script",
+    }
+    alerts.append(alert)
+
+    logging.info("Sending alerts to alert-manager...")
+    resp = requests.post(post_url, json=alerts)
+    resp.raise_for_status()
+    logging.info("Alerts sent to alert-manager.")
+
+def main():
+    PAI_URI = os.environ.get("PAI_URI")
+    certExpirationInfo = os.popen('kubeadm alpha certs check-expiration --config="/etc/kubernetes/kubeadm-config.yaml"').read()
```

Comment:
The command seems depends on kubeadm version. Refer https://kubernetes.io/docs/tasks/administer-cluster/kubeadm/kubeadm-certs/#check-certificate-expiration, the latest version don't need to add alpha in command.

One way to solve this is package kubeadm into docker image
Another way is check the cert manually. It's easy to use a lib to check cert expired time

Reply:
Done. use python OpenSSL.crypto.load_certificate(...) to read the crt file, and check the expired time.